pkg/agent/directive_handler_test.go

```diff
@@ -18,3 +19,33 @@ func TestHandleDirective(t *testing.T) {
 
 	dh.HandleDirective(ctx, a, d)
 }
+
+func TestHandleSecretSyncDirective(t *testing.T) {
+	// given
+	ctx := dlog.NewTestContext(t, false)
+
+	a := &agent.Agent{}
+	dh := &agent.BasicDirectiveHandler{}
+
+	d := &agentTypes.Directive{
```

Comment:
Some additional test coverage would be nice:

1. Set Action with multiple key/value pairs
2. Set Action with no key/value pairs
3. Set command with missing secret, namespace/name
3. Del Action for secret that exists
4. Del Action for secret that does not exist

I would also think some test with Roles missing as well.

Reply:
After looking at `secrets_test.go` some of these are already covered, correct?

Reply:
1. Data is passed directly to the function, so doesn't change anything, but I've updated the test case to reflect that.
2. Added test with an empty map, should be allowed though if empty. Not for nil (it will fail in the nil case)
3. It's handled by the directive handler.
4. There is a test case in secrets_test.
5. There is a test case in secrets_test. It doesn't return any error.